lib/standalone.js

```diff
@@ -187,6 +192,36 @@ async function standalone({
 
 	let stylelintResults;
 
+	{
+		/* extensions start */
+		const getConfigExtensions = async (/** @type {Config} */ c) => {
+			if (!c.extends) return c.extensions;
+
+			// see stylelint/stylelint#1993
+			const directory = configBasedir ? configBasedir : cwd;
+			const cfg = await extendConfig(stylelint, c, directory, directory);
+
+			return cfg.extensions;
+		};
+		const directories = fileList
+			.filter((p) => fs.existsSync(p) && fs.lstatSync(p).isDirectory())
+			.map((p) => path.relative(globCWD, p));
+		const cfgList = config ? await getConfigExtensions(config) : [];
+		const extList = cfgList?.length
+			? cfgList
+			: extensions?.length
+			? extensions
+			: DEFAULT_EXTENSIONS;
+
+		filePaths = filePaths.filter((p) => {
+			if (directories.some((directory) => p.startsWith(directory)))
+				return extList.some((extension) => p.endsWith(`.${extension}`));
+
+			return true;
+		});
+		/* extensions end */
+	}
```

Comment:
[suggest] Can we extract the block into a new function in this module like this?

```js
filePaths = filterFilePathsByExtensions(...);

if (filePaths.length) {
  // ...
}
```

Reply:
I could but it will have too many arguments for my taste.
I am not sure if it's for the best.